tests/v3_api/common.py

```diff
@@ -232,6 +232,40 @@ def validate_workload_with_sidekicks(p_client, workload, type, ns_name,
         assert "running" in pod["status"]["containerStatuses"][1]["state"]
 
 
+def validate_workload_paused(p_client, workload, expectedstatus):
+    workloadStatus = p_client.list_workload(uuid=workload.uuid)[0].paused
+    assert workloadStatus == expectedstatus
+
+
+def validate_pod_images(expectedimage, workload, ns_name):
+    pods = execute_kubectl_cmd("get pods -l workload.user.cattle.io/"
+                               "workloadselector=deployment-{0}-{1}"
+                               .format(ns_name, workload.name))
+    for pod in pods["items"]:
+        assert pod["spec"]["containers"][0]["image"] == expectedimage
+
+
+def validate_pods_are_running_by_id(expectedpods, workload, ns_name):
+    allpods = execute_kubectl_cmd("get pods -l workload.user.cattle.io/"
+                                  "workloadselector=deployment-{0}-{1}"
+                                  .format(ns_name, workload.name))
+    for expectedpod in expectedpods["items"]:
+        expectedpodname = expectedpod["metadata"]["name"]
+        for currentpod in allpods["items"]:
+            if currentpod["metadata"]["name"] == expectedpodname:
+                assert currentpod["status"]["phase"] == "Running"
```

Comment:
@adingilloRancher You need to make sure that every entry in the expectedpod is present in the currenpod list , if not fail. 
Also you can exit the for loop as soon as the match is found.